### PR TITLE
Changes speaker profile URL structure to “.../speaker/id/speakers-full-name” - fixes #78

### LINKED ISSUE #78: Speaker’s profile should have the following URL structure “.../speaker/speakers-full-name”
## Steps To Recreate
1. Go to home page
2. Click on “View Profile” for any of the speakers listed

## Work Needed
* [ ] Works on mobile
* [ ] Works on desktop
* [ ] Wrote some tests

### PR DESCRIPTION
**Pull request by srhboo**
- Route for speaker profile now matches structure “.../speaker/[id]/[firstname]-[lastname]”
- Name in URL is optional -> if the name is missing or does not match speaker data 'firstname-lastname', it is redirected to above (in the case of inaccurate deep links) 
- Updates link on speaker card to match above
- Some formatting updates from project prettier settings

Fixes #78

### CODE REVIEW

File: app/config/routes.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const routes = (
           <Route path="/get-started/profile" component={Profile} />
           <Route path="/get-started/work" component={Work} />
           <Route path="/get-started/social" component={Social} />
-          <Route path="/speaker/:id" component={Speaker} />
+          <Route path="/speaker/:id/:fullName?" component={Speaker} />
```
Comment (srhboo) [inline app/config/routes.js]: add optional route parameter (to allow redirect from just id)

File: app/pages/Home/components/SpeakerCard.js [hunk 2]
```diff
@@ -24,7 +24,9 @@ const SpeakerCard = ({ speaker }) => {
         <StyledButton
           color="primary"
           label="View profile"
-          href={`#/speaker/${speaker.id}`}
+          href={`#/speaker/${speaker.id}/${speaker.first_name}-${
+            speaker.last_name
+          }`}
```
Comment (srhboo) [inline app/pages/Home/components/SpeakerCard.js]: add name to link path

File: app/pages/Speaker/Speaker.js [hunk 3]
```diff
@@ -39,8 +39,8 @@ class SpeakerContainer extends Component {
   }
 
   componentDidMount() {
-    const { match: { params: { id } } } = this.props;
-    this.props.getSpeaker(id);
+    const { match: { params: { id, fullName } } } = this.props;
+    this.props.getSpeaker(id, fullName);
```
Comment (srhboo) [inline app/pages/Speaker/Speaker.js]: pass the full name param from url

File: app/redux/modules/speaker.js [hunk 4]
```diff
@@ -38,31 +40,35 @@ export function updateSelection(selected) {
...
   return dispatch => {
     axios
       .get(`${BASE_URL_PATH}/api/v1/profiles/${id}`)
       .then(res => {
         dispatch(updateSpeaker(res.data));
+
+        const { id, first_name, last_name } = res.data;
+        if (!equals(fullName, `${first_name}-${last_name}`)) {
+          dispatch(push(`/speaker/${id}/${first_name}-${last_name}`));
+        }
```
Comment (srhboo) [inline app/redux/modules/speaker.js]: if current path doesn't have proper structure (id/firstname-lastname), redirect to correct path

**Comment by MattGiesa** [comment]
Great work! It may be worth creating a function that takes first/last name and returns a url safe value, then we can use that to generate the url and also in the `if (!equals(...))` check you're doing in `app/redux/modules/speaker.js`. The function could also handle nulls/etc in one place.

**Comment by s-kennedy** [comment]
I'll second Matt's comment, I think we need to `encodeURIComponent()` the first and last names, in case there are spaces or special characters in the names.

**Comment by srhboo** [comment]
@MattGiesa @s-kennedy the speaker names are now sanitized (I had already started when I saw the comment about `encodeURIComponent()` but let me know if you think it's better to use that)
- Everything that's not A-Z a-z is removed, whitespaces are replaced by dashes (in case of additional names ex- middle name) 
- Some helper utils added, hopefully they are clear enough

Let me know if any more changes are needed!

**Pull request merged**